routers/web/repo/issue.go

```diff
@@ -1927,7 +1927,7 @@ func ViewIssue(ctx *context.Context) {
 			if pull.CanAutoMerge() || pull.IsWorkInProgress(ctx) || pull.IsChecking() {
 				return false
 			}
-			if (ctx.Doer.IsAdmin || ctx.Repo.IsAdmin()) && prConfig.AllowManualMerge {
+			if ctx.Repo.CanWrite(unit.TypeCode) && prConfig.AllowManualMerge {
```

Comment:
This is not right. At least, we need to check two functions
`pull_service.IsUserAllowedToMerge(ctx, pull, perm, ctx.Doer)` and `pb.IsUserMergeWhitelisted`, if it's not admin.

Reply:
I agree with lunny on this